Samples/AcnDataMatrix/MainWindow.xaml.cs

```diff
@@ -110,9 +110,50 @@ private void btnStartApp_Click(object sender, RoutedEventArgs e)
             aTimer.Start();            
         }
 
+        /// <summary>
+        /// A list of the Sync IP address that we have seen. We wait for a sync packet from all of these before rendering
+        /// </summary>
+        List<string> SyncIps = new List<string>();
+
+        /// <summary>
+        /// A list of IP addresses of nodes that we have seeen sync packets from since the last render
+        /// </summary>
+        List<string> SeenIps = new List<string>();
+
+        /// <summary>
+        /// Called when a new sync packet is recieved
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
         private void Socket_NewSynchronize(object sender, NewPacketEventArgs<StreamingAcnSynchronizationPacket> e)
         {
-            example.Draw(TimeSpan.Zero);
+            string sourceIp = e.Source.Address.ToString();
+
+            Dispatcher.BeginInvoke(new Action(() =>
+            {
+                if (!SyncIps.Contains(sourceIp))
+                    SyncIps.Add(sourceIp);
+
+                if (!SeenIps.Contains(sourceIp))
+                    SeenIps.Add(sourceIp);
+
+                if (SeenIps.Count == SyncIps.Count)
+                {
+                    example.Draw(TimeSpan.Zero);
+                    SeenIps = new List<string>();
```

Comment:
Do `SeenIps.Clear()` to avoid reallocating unnecessarily; this will maintain the hashet's capacity.